root/static/scripts/tests/utility/isGuid.js

```diff
@@ -31,6 +31,11 @@ test('isGuid', function (t) {
 
   t.ok(
     isGuid('89ad4ac3-39f7-470e-963a-56509c546377'),
-    'Valid MBID is a GUID',
+    'Valid MBID with lowercase letters is a GUID',
+  );
+
+  t.ok(
```

Comment:
This probably isn't specific to this change, but I noticed that this test file (and seemingly any tests after those for `isUselessMediumTitle`) doesn't seem get run when I load `/static/scripts/tests/web.html` on my local development MBS instance. Any idea why that is?

Here's the order in `root/static/scripts/tests/index-web.js`:

```js
require('./autocomplete.js');
require('./common/immutable-entities.js');
require('./Control/URLCleanup.js');
require('./CoverArt.js');
require('./edit.js');
require('./edit/utility/getUpdatedTrackArtists.js');
require('./edit/utility/isUselessMediumTitle.js');
require('./edit/utility/linkPhrase.js');
require('./entity.js');
require('./GuessCase.js');
require('./guessFeat.js');
require('./i18n.js');
...
```

And here's the bottom of the page:

![image](https://github.com/metabrainz/musicbrainz-server/assets/40385/3ea52a34-cf6e-44cc-a2d7-3b6373c488dc)

When I move `isGuid.js` to the top, I see its (successful) output. I didn't notice anything fishy in `isUselessMediumTitle.js` or `linkPhrase.js`.

Reply:
The fact that it ends before linkPhrase.js is suspicious, since that file depends on cached data from root/static/scripts/tests/typeInfo.js which may be out of date. You can try removing that file and re-compiling the JS tests:

```sh
rm root/static/scripts/tests/typeInfo.js; ./script/compile_resources.sh tests
```

Reply:
Hmm, that definitely seems related. I don't have a `root/static/scripts/tests/typeInfo.js` file, and I've been seeing (and ignoring) Flow errors related to that for a long time:

```
Error ------------------------------------------------------- root/static/scripts/tests/edit/utility/linkPhrase.js:22:45

Cannot resolve module `../../typeInfo.js`. [cannot-resolve-module]

   22| import {linkAttributeTypes, linkTypes} from '../../typeInfo.js';
                                                   ^^^^^^^^^^^^^^^^^^^
```

When I run `compile_resources.sh tests` within the `musicbrainz` Docker service, it also fails due to the missing file:

```
...
Skipping typeInfo.js dump; no running TEST database?
...
  ERROR in ./root/static/scripts/tests/autocomplete2.js 861:0-51
  Module not found: Error: Can't resolve './typeInfo.js' in '/musicbrainz-server/root/static/scripts/tests'
```

It's been a long time since I set this up, but I suspect that the issue may be that I (IIRC) followed the "Local development of MusicBrainz Server" instructions in the musicbrainz-docker README.md rather than the "Test setup" instructions. I think the main difference may be `admin/configure add musicbrainz-dev` rather than `admin/configure add musicbrainz-standalone`. Does that sound plausible?

Reply:
Actually, on second reading, the "Test setup" there seems like it's referring more to "running your own little server with sample data" than "running unit tests", so maybe not.

I also get/ignore a bunch of `musicbrainz_test` errors in the logs printed by `sudo docker-compose up`:

```
db_1           | 2024-04-15 20:19:55.486 UTC [78] FATAL:  database "musicbrainz_test" does not exist
db_1           | 2024-04-15 20:25:04.336 UTC [105] FATAL:  database "musicbrainz_test" does not exist
```

Any way to repair this, short of nuking everything? It's otherwise been working fine for development. Adding @yvanzo, and happy to file a new ticket if that'd be a better place for this.

Reply:
Thanks! I must've missed `script/create_test_db.sh` initially. Running that and then `script/compile_resources.sh tests` got all of the tests to run.